packages/flutter/lib/src/material/button.dart

```diff
@@ -315,9 +323,17 @@ class _RawMaterialButtonState extends State<RawMaterialButton> {
     return widget.elevation;
   }
 
+  ShapeBorder get _effectiveShape {
```

Comment:
Could be `MaterialProperty.resolve<ShapeBorder>(...)`; see comment below

Reply:
Done.